service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -231,14 +232,22 @@ public ResponseEntity<RunSetStateResponse> postRunSet(RunSetRequest request) {
   @Override
   public ResponseEntity<AbortRunSetResponse> abortRunSet(UUID runSetId) {
     AbortRunSetResponse aborted = new AbortRunSetResponse();
-    List<UUID> dummyRunsList = new ArrayList<>();
+    aborted.runSetId(runSetId);
 
-    dummyRunsList.add(UUID.randomUUID());
-    dummyRunsList.add(UUID.randomUUID());
+    List<Run> runningWorkflows =
+        runDao.getRuns(new RunDao.RunsFilters(runSetId, NON_TERMINAL_STATES));
+    List<String> submittedAbortWorkflows = new ArrayList<>();
 
-    aborted.runSetId(runSetId);
+    for (Run run : runningWorkflows) {
+      try {
+        cromwellService.submitAbortWorkflow(run);
+        submittedAbortWorkflows.add(run.engineId());
+      } catch (cromwell.client.ApiException e) {
+        log.error("Unable to abort workflow %s".formatted(run.engineId()), e);
+      }
+    }
+    aborted.runs(submittedAbortWorkflows);
```

Comment:
I think we should also mark the Run Ids in `Canceling` state for which calling the abort endpoint was successful. For Run Ids where we received the ApiException we shouldn't mark them in `Canceling` state.

Related to this, the Run Set should also be marked in `Canceling` state in database.

Reply:
Does this happen automatically in the `SmartRunsPoller`? When I testing this in CBAS-UI, the status automatically went from `Canceling` to `Canceled` for both run sets and runs by itself.

Reply:
Ah right, that would be because the UI was polling for status updates every 1 min and Cromwell was returning `Canceling` state for each run. But I think it would be better to also mark the Runs here itself in Canceling state. CBAS should not rely on UI to poll for status update of Runs to have the Runs transition to Canceling state.

And since Cromwell doesn't know about Run Sets, we should mark the Run Set in Canceling state here too.